File: OCR-Translate/Models/TranslationResult.swift
import Foundation

struct TranslationResult: Equatable {
    let originalText: String
    let translatedText: String
    let sourceLanguage: String
    let targetLanguage: String
    let timestamp: Date

    init(
        originalText: String,
        translatedText: String,
        sourceLanguage: String = "ja",
        targetLanguage: String = "zh-CN",
        timestamp: Date = Date()
    ) {
        self.originalText = originalText
        self.translatedText = translatedText
        self.sourceLanguage = sourceLanguage
        self.targetLanguage = targetLanguage
        self.timestamp = timestamp
    }
}

File: win/NGMproofread.Windows/Models/TranslationResult.cs
namespace NGMproofread.Windows.Models;

public class TranslationResult
{
    public string OriginalText { get; init; } = "";
    public string TranslatedText { get; init; } = "";
    public string SourceLanguage { get; init; } = "ja";
    public string TargetLanguage { get; init; } = "zh-CN";
    public DateTime Timestamp { get; init; } = DateTime.Now;
}
